code/espurna/sensors/VL53L1XI2CSensor.h

```diff
@@ -0,0 +1,86 @@
+// -----------------------------------------------------------------------------
+// VL53L1X Sensor over I2C
+// Copyright (C) 2017-2018 by Xose Pérez <xose dot perez at gmail dot com>
+// -----------------------------------------------------------------------------
+
+#if SENSOR_SUPPORT && VL53L1X_SUPPORT
+
+#pragma once
+
+#undef I2C_SUPPORT
+#define I2C_SUPPORT 1 // Explicitly request I2C support.
+
+
+#include "Arduino.h"
+#include "I2CSensor.h"
+#include "VL53L1X.h"
+
+class VL53L1XI2CSensor : public I2CSensor {
+
+    public:
+
+        // ---------------------------------------------------------------------
+        // Public
+        // ---------------------------------------------------------------------
+
+        VL53L1XI2CSensor(): I2CSensor() {
+            _count = 1;
+            _sensor_id = SENSOR_VL53L1X_ID;
+        }
+
+        // ---------------------------------------------------------------------
+        // Sensor API
+        // ---------------------------------------------------------------------
+
+        // Initialization method, must be idempotent
+        void begin() {
+            // I2C auto-discover
+            unsigned char addresses[] = {0x29};
+            _address = _begin_i2c(_address, sizeof(addresses), addresses);
+            if (_address == 0) return;
+
+            _sensor = new VL53L1X();
+            _sensor->setDistanceMode(VL53L1X_DISTANCE_MODE);
+            _sensor->setMeasurementTimingBudget(VL53L1X_MEASUREMENT_TIMING_BUDGET);
+            _sensor->startContinuous(VL53L1X_INTER_MEASUREMENT_PERIOD);
+
+            _ready = true;
+        }
+
+        // Descriptive name of the sensor
+        String description() {
+            char buffer[25];
+            snprintf(buffer, sizeof(buffer), "VL53L1X @ I2C (0x%02X)", _address);
+            return String(buffer);
+        }
+
+        // Descriptive name of the slot # index
+        String slot(unsigned char index) {
+            return description();
+        };
+
+        // Type for slot # index
+        unsigned char type(unsigned char index) {
+            if (index == 0) return MAGNITUDE_DISTANCE_MILLIMETERS;
+            return MAGNITUDE_NONE;
+        }
+
+        // Pre-read hook (usually to populate registers with up-to-date data)
+        void pre() {
+            _error = SENSOR_ERROR_OK;
+        }
+
+        // Current value for slot # index
+        double value(unsigned char index) {
+            if (index != 0) return 0;
+
+            return _sensor->read();
```

Comment:
@xoseperez this is a blocking call. I am not sure if espurna supports asyncs events. I also have no idea on best practices..